### PR TITLE
Aded support for HostMask user mode (+x)

### LINKED ISSUE #43: Host Hiding (+x)
Implement [Host Hiding](https://defs.ircdocs.horse/defs/usermodes.html#x-host-hiding) (`+x`)

This should:

* Set `+x` when a user connects
* Which hides their real host from others (*currently a stupid/dumb SHA255 hash; See #31*)
* Users are allowed to turn this off; so it's not a "flag".

### PR DESCRIPTION
**Pull request by kzisme**
Adds support for HostMask usermode (+x)

Closes #43

### CODE REVIEW

File: irc/modes.go [hunk 1]
```diff
@@ -58,11 +58,12 @@ const (
 	Registered UserMode = 'r' // not a real user mode (flag)
 	SecureConn UserMode = 'z'
 	SecureOnly UserMode = 'Z'
+    HostMask   UserMode = 'x'
```
Comment (prologic) [inline irc/modes.go]: Please fix your editor to invoke `go fmt` or run it manually.

**Comment by prologic** [comment]
So far so good. This should add the new mdoe; but now we need the check:
  
  if user has HostMask:
    hide their host from /whois /who
  else:
    show their host in /whois /who

**Comment by kzisme** [comment]
Also adding 3x test 

- +x
- -x
- +x default

**Comment by prologic** [comment]
Updated PR title and desc to automatically close #43 when landed

File: main_test.go [hunk 2]
```diff
@@ -235,6 +235,30 @@ func TestChannel_InviteOnly(t *testing.T) {
...
+	client.AddCallback("MODE", func(e *irc.Event) {
+		actual <- e.Message()
+	})
+
+	defer client.Quit()
+
+	select {
+	case res := <-actual:
+		assert.Equal(expected, res)
+	}
```
Comment (prologic) [inline main_test.go]: Don't forget the timtout case:

```
	case <-time.After(TIMEOUT):
		assert.Fail("timeout")
	}
```

File: main_test.go [hunk 3]
```diff
@@ -235,6 +235,30 @@ func TestChannel_InviteOnly(t *testing.T) {
...
+	client.AddCallback("001", func(e *irc.Event) {
+		client.Mode(client.GetNick(), "MODE +x"+client.GetNick())
+	})
+
+	client.AddCallback("MODE", func(e *irc.Event) {
+		actual <- e.Message()
+	})
+
+	defer client.Quit()
+
```
Comment (prologic) [inline main_test.go]: Don't forget to start the client:

```
go client1.Loop()
```

File: main_test.go [hunk 4]
```diff
@@ -235,6 +235,30 @@ func TestChannel_InviteOnly(t *testing.T) {
...
+func TestUser_HostMask(t *testing.T) {
+	assert := assert.New(t)
+
+	client := newClient(false)
+
+	expected := client.GetNick() + " sets mode to +x"
+	actual := make(chan string)
+
+	client.AddCallback("001", func(e *irc.Event) {
+		client.Mode(client.GetNick(), "MODE +x"+client.GetNick())
```
Comment (prologic) [inline main_test.go]: This is not right.

See: https://godoc.org/github.com/thoj/go-ircevent#Connection.Mode

File: irc/client.go [hunk 5]
```diff
@@ -218,6 +218,7 @@ func (client *Client) Register() {
 		return
 	}
 	client.registered = true
+	client.flags[HostMask]
```
Comment (prologic) [inline irc/client.go]:
```#!go
	client.flags[HostMask] = true
```

File: main_test.go [hunk 6]
```diff
@@ -235,6 +236,82 @@ func TestChannel_InviteOnly(t *testing.T) {
...
+	assert := assert.New(t)
+
+	client1 := newClient(false)
+	client2 := newClient(false)
+
+	expected := fmt.Sprintf("%x", sha256.Sum256([]byte("localhost")))
+	actual := make(chan string)
+
+	client1.AddCallback("001", func(e *irc.Event) {
+		client1.Mode(client1.GetNick(), "+x")
```
Comment (prologic) [inline main_test.go]: What's with the `+` here? `-x+`?

Comment (kzisme) [reply main_test.go]: @prologic I had that added in locally (when trying to get the test suite to pass) when running `go test -d` it runs them all with a pass, but travis fails unless I add the 'x' to '-x+'....

**Pull request merged**